File: 2023-2024/sourcecode/mod2/lab9/RepositoryBasicsJava/app/src/main/java/com/example/repositorybasicsjava/ui/UserComparator.java
package com.example.repositorybasicsjava.ui;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;

import com.example.repositorybasicsjava.model.User;

public class UserComparator extends DiffUtil.ItemCallback<User> {
    @Override
    public boolean areItemsTheSame(@NonNull User oldItem, @NonNull User newItem) {
        return oldItem == newItem;
    }

    @Override
    public boolean areContentsTheSame(User oldItem, @NonNull User newItem) {
        return oldItem.equals(newItem);
    }
}

File: 2023-2024/sourcecode/mod2/lab9/RepositoryBasicsKotlin/app/src/main/java/com/example/repositorybasicskotlin/ui/UserComparator.kt
package com.example.repositorybasicskotlin.ui

import androidx.recyclerview.widget.DiffUtil
import com.example.repositorybasicskotlin.model.User

class UserComparator : DiffUtil.ItemCallback<User>() {
    override fun areItemsTheSame(oldItem: User, newItem: User): Boolean {
        return oldItem === newItem
    }

    override fun areContentsTheSame(oldItem: User, newItem: User): Boolean {
        return oldItem == newItem
    }
}